code/controllers/subsystem/dbcore.dm

```diff
@@ -62,68 +51,66 @@ SUBSYSTEM_DEF(dbcore)
 	return ..()
 
 /datum/controller/subsystem/dbcore/proc/Connect()
-	if(IsConnected())
-		return TRUE
-
-	if(failed_connection_timeout <= world.time) //it's been more than 5 seconds since we failed to connect, reset the counter
-		failed_connections = 0
-
-	if(failed_connections > FAILED_DB_CONNECTION_CUTOFF)	//If it failed to establish a connection more than 5 times in a row, don't bother attempting to connect for 5 seconds.
-		failed_connection_timeout = world.time + 50
-		return FALSE
-
 	if(!CONFIG_GET(flag/sql_enabled))
 		return FALSE
-
+	if(SSdbcore.IsConnected())
+		return TRUE
 	var/user = CONFIG_GET(string/feedback_login)
 	var/pass = CONFIG_GET(string/feedback_password)
 	var/db = CONFIG_GET(string/feedback_database)
 	var/address = CONFIG_GET(string/address)
 	var/port = CONFIG_GET(number/port)
+	var/list/connection_result = json_decode(rustg_sql_connect_pool(address, port, user, pass, db, "", "", ""))

```

Comment:
golfball you're a fucking idiot you should be passing the BSQL config like max_connections and timeout to this

Reply:
we don't use BSQL anymore dude

Reply:
^ actual state of goof

Reply:
Tomorrow: 
![](https://qph.fs.quoracdn.net/main-qimg-4e5c6854fff25712afa25f5240fbc660)